diff --git a/node_modules/@graphql-tools/schema/types.d.ts b/node_modules/@graphql-tools/schema/types.d.ts
index ec40a77..7065f69 100644
--- a/node_modules/@graphql-tools/schema/types.d.ts
+++ b/node_modules/@graphql-tools/schema/types.d.ts
@@ -1,5 +1,4 @@
-import { TypeSource, IResolvers, IResolverValidationOptions, GraphQLParseOptions, PruneSchemaOptions } from '@graphql-tools/utils';
-import { SchemaExtensions } from '@graphql-tools/merge';
+import { TypeSource, IResolvers, IResolverValidationOptions, GraphQLParseOptions, PruneSchemaOptions, SchemaExtensions } from '@graphql-tools/utils';
 import { BuildSchemaOptions } from 'graphql';
 /**
  * Configuration object for creating an executable schema
