diff --git a/node_modules/relay-test-utils-internal/lib/testschema.graphql b/node_modules/relay-test-utils-internal/lib/testschema.graphql
index ce0e0f3..1071e39 100644
--- a/node_modules/relay-test-utils-internal/lib/testschema.graphql
+++ b/node_modules/relay-test-utils-internal/lib/testschema.graphql
@@ -1008,6 +1008,7 @@ type User implements Named & Node & Actor & HasJsField & Entity & AllConcreteTyp
     last: Int
     orderby: String
   ): CommentsConnection
+  count: Int!
   doesViewerLike: Boolean
   emailAddresses: [String]
   environment: Environment
