diff --git a/node_modules/relay-test-utils-internal/lib/testschema.graphql b/node_modules/relay-test-utils-internal/lib/testschema.graphql
index 2b6036e..4536218 100644
--- a/node_modules/relay-test-utils-internal/lib/testschema.graphql
+++ b/node_modules/relay-test-utils-internal/lib/testschema.graphql
@@ -496,7 +496,7 @@ interface FeedUnit {
 
 type FriendsConnection {
   count: Int
-  edges: [FriendsEdge]
+  edges: [FriendsEdge]!
   pageInfo: PageInfo
 }
 
@@ -1024,6 +1024,7 @@ type User implements Named & Node & Actor & HasJsField & Entity & AllConcreteTyp
     last: Int
     orderby: String
   ): CommentsConnection
+  count: Int!
   doesViewerLike: Boolean
   emailAddresses: [String]
   environment: Environment
