diff --git a/node_modules/source-map-js/source-map.d.ts b/node_modules/source-map-js/source-map.d.ts
index 9f8a4b3..2117555 100644
--- a/node_modules/source-map-js/source-map.d.ts
+++ b/node_modules/source-map-js/source-map.d.ts
@@ -5,7 +5,7 @@ declare module 'source-map-js' {
     }
 
     export interface RawSourceMap extends StartOfSourceMap {
-        version: string;
+        version: number;
         sources: string[];
         names: string[];
         sourcesContent?: string[];
@@ -75,6 +75,7 @@ declare module 'source-map-js' {
         setSourceContent(sourceFile: string, sourceContent: string): void;
         applySourceMap(sourceMapConsumer: SourceMapConsumer, sourceFile?: string, sourceMapPath?: string): void;
         toString(): string;
+        toJSON(): RawSourceMap;
     }
 
     export interface CodeWithSourceMap {
